pkg/controller/common/helpers.go

```diff
@@ -658,3 +658,60 @@ func dedupePasswdUserSSHKeys(passwdUser ign2types.PasswdUser) ign2types.PasswdUs
 
 	return passwdUser
 }
+
+// CalculateConfigFileDiffs compares the files present in two ignition configurations and returns the list of files
+// that are different between them
+func CalculateConfigFileDiffs(oldIgnConfig, newIgnConfig *ign3types.Config) []string {
```

Comment:
WDYT about returning a struct type with the diff that distinguishes add/remove/modify?

This is like https://github.com/coreos/bootupd/blob/499e2af364dee4bbc9811c62ddfd001b4686cd9d/src/filetree.rs#L46

Reply:
Do we need that info somewhere in MCO?

Reply:
We could use it for `isDrainRequired` (see https://github.com/openshift/machine-config-operator/pull/2851)

Reply:
This can be at least useful to log messages by caller instead of doing it here,in the context of https://github.com/openshift/machine-config-operator/pull/2870/files#r768161851

Reply:
@jkyros wdyt? Time to pull out the howitzer or not?

Reply:
I think given this is needed for https://github.com/openshift/machine-config-operator/pull/2851 which I think we want to get in before Friday, it might be better to fix this later if we decide it's worth it

Reply:
I think we end up having to someday, but that day doesn't necessarily have to be today if there are timeline concerns.  
- Any half-measure we take is just going to end up here again later
- This would let us keep our debug logging without forcing it on all function consumers.  
- I don't know how much of the utility here you need/want for #2851 ?
- I do think I can rebase in time on #2802 if that was a concern

Reply:
sure, let's create a card for this in Jira so that someone pick this up later on.